packages/@icon-magic/icon-models/src/icon.ts

```diff
@@ -189,4 +194,11 @@ export class Icon {
     // return the object
     return config;
   }
+
+  /**
+   * Sets the generated metadata passed in
+   */
+   setGeneratedMetadata(generatedMetadata: GeneratedMetadata) {
```

Comment:
I would create a function to addGeneratedMetadata and pass in the object to add rather than having to set the whole object each time

Reply:
Thats a great point. Thanks for pointing it out! I have changed to `add` instead of `set`.